[com_joomlaupdate] Allow a different reinstall URL than the update URL

zero-24 opened the pull request:
Pull Request for Issue #11488 
## Notice

Don't do that **never** on your Live Site please install a dev / test install!
#### Summary of Changes

With this patch applyed you can use a different reinstall URL than the update url.

For Updating you need just the `Joomla_3.6.x_to_3.6.2-Stable-Patch_Package.zip` package for reinstall you need the `Joomla_3.6.2-Stable-Update_Package.zip`.

So now if we have a reinstall URL configured lets use it.
#### Testing Instructions

Tip: Do not hit the `Update` or `Reinstall` button on testing else the applyed PR gets reverted!
- install 3.6.2
- go to com_joomlaupdate
- see the reinstall URL is `Joomla_3.6.2-Stable-Update_Package.zip`
- apply this patch
- switch the update server to http://www.jah-tz.de/downloads/core/list2.xml
- go to com_joomlaupdate again
- see the reinstall URL is `Joomla_3.6.x_to_3.6.2-Stable-Patch_Package.zip`
- switch back to the core Update server
- see that now we have  `Joomla_3.6.2-Stable-Update_Package.zip` back.
- switch the update server to http://www.jah-tz.de/downloads/core/list3.xml
- notice there is a update that points to `Joomla_3.6.2-Stable-Update_Package.zip`
#### Documentation Changes Required

There is a `core used only` new entry for the extension.xml used by the core updater.
#### Additional Information

The Joomla Core Update Server needs to be extended to use the function with this line. if that line is missing reinstallurl is the same as the update url:

```
<reinstallurl type="full" format="zip">https://github.com/joomla/joomla-cms/releases/download/3.6.2/Joomla_3.6.2-Stable-Update_Package.zip</reinstallurl>
```

A complete example:

```
<?xml version="1.0" ?>
<updates>
    <update>
        <name>Joomla! 3.6</name>
        <description>Joomla! 3.6 CMS</description>
        <element>joomla</element>
        <type>file</type>
        <version>3.6.2</version>
        <infourl title="Joomla!">https://www.joomla.org/announcements/release-news/5667-joomla-3-6-2-released.html</infourl>
        <downloads>
            <downloadurl type="full" format="zip">https://github.com/joomla/joomla-cms/releases/download/3.6.2/Joomla_3.6.x_to_3.6.2-Stable-Patch_Package.zip</downloadurl>
            <reinstallurl type="full" format="zip">https://github.com/joomla/joomla-cms/releases/download/3.6.2/Joomla_3.6.2-Stable-Update_Package.zip</reinstallurl>
        </downloads>
        <tags>
            <tag>stable</tag>
        </tags>
        <maintainer>Joomla! PLT</maintainer>
        <maintainerurl>https://www.joomla.org</maintainerurl>
        <section>STS</section>
        <targetplatform name="joomla" version="3.6" />
        <php_minimum>5.3.10</php_minimum>
    </update>
</updates>
```

zero-24 committed:
allow reinstll to be a different url

zero-24 committed:
some more places to fix

zero-24 committed:
typo

zero-24 committed:
travis my friend :)

brianteeman commented:
Why would we want to do this - what is the use case<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11497">issues.joomla.org/joomla-cms/11497</a>.</sub>

zero-24 commented:
On request by @jeckodevelopment and @wilsonge so there is no need to update with a 10mb package from 3.6.1 to 3.6.2 ;)

brianteeman commented:
@zero-24 sorry if I misunderstood but I read this as allowing an install from a different domain

jeckodevelopment commented:
@brianteeman it does, but only to test this PR since we cannot use the official release URL to test the PR.

brianteeman commented:
that cannot be in the final version then -

On 7 August 2016 at 14:25, Luca Marzo <notifications@github.com> wrote:

> @brianteeman <https://github.com/brianteeman> it does, but only to test
> this PR since we cannot use the official release URL to test the PR.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/11497#issuecomment-238082225>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABPH8fdKP7MGF6KdwxvqOVFbzuXVpgztks5qddyygaJpZM4JeT8m>
> .
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

jeckodevelopment commented:
it should be fixed after tests.

AlexRed commented:
In Joomla 3.6.1 after patch applyed and switch the update server (http://www.jah-tz.de/downloads/core/list2.xml) the com_joomlaupdate show the wrong package URL Joomla_3.6.x_to_3.6.2-Stable-Patch_Package.zip  and not the correct Joomla_3.6.1_to_3.6.2-Stable-Patch_Package.zip 

In Joomla 3.6.0 after patch applyed and switch the update server (http://www.jah-tz.de/downloads/core/list2.xml) the com_joomlaupdate show the correct package URL Joomla_3.6.x_to_3.6.2-Stable-Patch_Package.zip

zero-24 commented:
That is the expected behavior. The 3.6.1 to 3.6.2 package is not implemented yet. @alexred

AlexRed commented:
ok, I can test it when implemented

zero-24 commented:
There is no plan to implement that here as this can be done with the current code just with another update definition ;)

yvesh commented on administrator/components/com_joomlaupdate/views/default/tmpl/default.php:
```diff
@@ -50,9 +50,6 @@
 		<?php endif; ?>
 
-		<input type="hidden" name="task" value="update.download" />
-		<input type="hidden" name="option" value="com_joomlaupdate" />
```
@zero-24 You could have kept the option here, as it is the same for both templates :-)

zero-24 replied:
No. The update template uses download and the reinstall template use reinstall ;)

zero-24 replied:
Ok double checked. Yes i'm ging to fix this if im back on my pc :)

zero-24 committed:
move input option

zero-24 commented:
can we have the labels here @joomla-cms-bot ?

zero-24 committed:
Merge branch 'staging' into improveReinstall

brianteeman commented:
@zero-24 bot not working so brian-bot did it

zero-24 committed:
Merge branch 'staging' into improveReinstall

zero-24 committed:
Merge branch 'staging' into improveReinstall

zero-24 committed:
Merge branch 'staging' into improveReinstall

zero-24 committed:
Merge branch 'staging' into improveReinstall

zero-24 committed:
Merge branch 'staging' of github.com:/joomla/joomla-cms into improveReinstall

zero-24 committed:
use __DEPLOY_VERSION__

zero-24 committed:
Merge branch 'staging' into improveReinstall

zero-24 committed:
Merge branch 'staging' into improveReinstall

zero-24 committed:
Merge branch 'staging' into improveReinstall

zero-24 committed:
Merge branch 'staging' into improveReinstall

zero-24 committed:
Merge branch 'staging' into improveReinstall

zero-24 committed:
Merge branch 'staging' into improveReinstall

zero-24 committed:
Merge branch 'staging' into improveReinstall

zero-24 committed:
Merge branch 'staging' into improveReinstall

zero-24 commented:
Is there anything missing you need to test this or is there no usecase for that feature?

zero-24 committed:
Merge branch 'staging' into improveReinstall

zero-24 committed:
Merge branch 'staging' into improveReinstall

zero-24 committed:
Merge branch 'staging' into improveReinstall

zero-24 committed:
Merge branch 'staging' into improveReinstall

zero-24 committed:
Merge branch 'staging' into improveReinstall

zero-24 committed:
Merge branch 'staging' into improveReinstall

zero-24 committed:
Merge branch 'staging' into improveReinstall

zero-24 committed:
Merge branch 'staging' into improveReinstall

zero-24 committed:
Merge branch 'staging' into improveReinstall

zero-24 committed:
Merge branch 'staging' into improveReinstall

zero-24 committed:
Merge branch 'staging' into improveReinstall

zero-24 committed:
Merge branch 'staging' into improveReinstall

zero-24 commented:
So is there now reason to improve the update prozess? Just because this (or similiar) is not merged we still need to push everytime the stable update package that can break installations because of the timeout problems.

I can explain anything you need this to be tested just give me a line what do you need to be explained ;)

zero-24 committed:
Merge branch 'staging' into improveReinstall

Commit:
Correct spelling (#12564)

photodude committed:
Remove JFilter mock (#12484)

We don't need JFilter hanging out in the JInstaller Adapter tests

Commit:
Stop allowing failures on PHP 7.1 (#12599)

andrepereiradasilva committed:
Improvements in the Associations Edit Workflow (#12344)

* changes in model preprocessForm to allow all associations fields to be dumped

* adjustement

* there is no need to block the language field now

* simplify association tab in the templates files

* let's make the layout more dynamic

* add the js behaviours

* solve one of strange bug ...

* improvements and allow associations on create

* correct language string

* render a warning message

* solve categoies issue

* add new language methods getContentLanguages and getInstalledLanguages

* let's now load all content languages, including the unpublished ones

* correct label for bug in categories modal field! all other model fields have this

* ups correction in js

* default pivot is language code

* doc blocks corrections

* cs

* language correction thanks brian

* fix conflicts final

photodude committed:
When $this->cursor is false, throw an Exception (#12359)

A call to pg_result_error_field() when $this->cursor is false from a query failing will result in an error `pg_result_error_field() expects parameter 1 to be resource, boolean given`

Try to get a valid error message and throw a JDatabaseExceptionExecuting Exception

matrikular committed:
Implement a check / notification for an invalid or missing access file (#12608)

* Implement a check / notification for an invalid or missing access file

* Add language string for invalid or missing access file notification

* Add language string for invalid or missing access file notification

* Reduce log message as suggested by mbabker

* Reduce log message for #12608 as suggested by mbabker

* Reduce log message for #12608 as suggested by mbabker

* Change string to match actual translation text

ciar4n committed:
Grid within well changed to row-fluid (#11995)

brianteeman committed:
update doc block (#12569)

brianteeman committed:
Folder icons in front end (#12551)

* Add folder size to protostar .less

* compiled protostar

brianteeman committed:
rename folder value (#12579)

infograf768 committed:
[3.7.x] Regression Hathor modules and users manager (#12412)

andrepereiradasilva committed:
Native language (adjustments after 12036 merge) (#12394)

* restore native language

* remvoe space

* correct

* simplify

* improve code

* ups

* native language in unit test too

alex7r committed:
Add detachObserver (#11037)

* Add detachObserver

Adds ability to detach observers

* Simple CS fix

csthomas committed:
3.6.4 Sample data: Escape quota in json data (#12501)

phproberto committed:
[fix] new extensions not installed on already existing packages (#12545)

Commit:
$menu->params is already a Registry object. (#12557)

eshiol committed:
Overridden strings in both back-end and front-end (#12576)

phproberto committed:
[imp] remove hardcoded published column (#12585)

Commit:
Fix JUri path in language filter plugin (#12600)

laoneo committed:
Custom fields (#11833)

* Update gallery.xml

* Update fields.php

* Update internal.php

* Update field.php

* Update dprules.php

* Update section.php

* Update base.php

* Update view.html.php

* Update view.html.php

* Update render.php

* Update modal_article.php

* Update sql.php

* Update fields.php

* Remove commented debug code

* Update fields.php

* #14 fixed Joomla Coding Standards and added function dock blocks code

* #14 fixed Joomla Conding Standards and added function dock blocks

* #14 Fixed class dock block mistake

* we did not require this any more

* #14 fixed Joomla Coding Standards and added function dock block in helper file at back-end

* #14 Fixed only Joomla Coding Standards in FieldsHelper file

* #14 removed where function, added dock blocks and fixed dock bocks coding standards errors

* #14 formatted as the function doc blocks

* #14 formatted as the function doc blocks

* Check the active menu correct

Closes #72

* Move the label field to the options tab

Closes #22

* Label has it's own field and must not stay in params

Regression from #22

* Backport from Digital-Peak/DPFields#32

* Fix the save and copy action

Closes #37

* Store data on reload

Closes #76

* Don't add always the All language to the query

Closes #38

* Count the items in the category manager from com_fields

Closes #61

* Corrected colspan when fields are assigned to a language

* #14 fixed conflict

* Don't save the All category

Closes #19

* Minor language updates

* Fix "Show On" tooltip's description

The tooltip's description for the 'Show On' field is not showing because it is currently looking for the label.

* Don't hardcode com_content.article

Closes #80

* Load the types form when the data is an array

* Create the links correct for the count item feature

Closes #82

* Show the value from com_fields instead of the the users helper

Closes #81

* Set the language correct on the fields cache

Closes #47

* Removed string conversion error

Closes #47

* Changed links to 3.7

* Remove legacy DPFields code

* Get data from state correctly

* Add joins to the com_fields in search plugin (#83)

Add joins to com_fields in the content search plugin

* Add new core rule core.edit.value (#75)

* Add 'onFieldBeforePrepare' and 'onFieldAfterPrepare' events

* Rename "Field Category" to "Field Group"

Closes #91

* Add option to disable custom fields per component

Closes #87

* Use select box to define the images directory

Closes #97

* Truncate correctly

Closes #101

* Change description field to plain textarea

Closes #94
CLoses #99

* Parameters are more clear and improved comment

Adapted change from
https://github.com/joomla/joomla-cms/pull/10722/commits/32c9048b88383ba6028743a48517176952d9a574

* onFieldBeforePrepare and onFieldAfterPrepare events now triggered even when custom field value is empty

* Cast before access

* Create params even when they are empty

* Add com_fields menu entries on back end

Closes #109

* Extract the context correctly

Closes #112

* Add custom fields for contact

* Add inline group create

Closes #114

* Add simple context mapping

Closes #117

* Dont assign a not existing category when a field is saved with no catid

* Rename string with group and not category

Closes #124

* Category strings should be named Field Groups

Closes #126

* Show the fields on the blog listing

Closes #108

* Installer crashes somehow

Closes #132

* Merge the types set up into JFormField (#104)

* Merge field types into JFormFields

* Converting the rest of the fields

* Merge DP Field Rendering with Joomla Form Fields

* Remove the gallery scripts

* Rename the interface to JFormDomfieldinterface for auto loading

* Add article field as demo how a component can add fields

* Removing mustache, will be replaced later with a new parser

See #12 for the discussion

* Fixing composer setup, reverting to 3.6.x branch

* Fixing code style errors

Closes #14

* Rename upgrade file to 3.7

* Strip slashes from description

Related to #139

* Removing Simplepie from the database installer script

Reverts 8381e46758d18c6286fde35c52b1a2a475d7d0df
See comments for more information

* Show field group description in form

Closes #139

* List field changed key to value and value to name

Closes #121

* Use the first category as field filter when creating an article

Closes #144

* Dont show a user field on the front end

Closes #149

* Add always the fields path when adding the custom fields

Closes #150

* Added hint (placeholder) option

Closes #151

* Correct redirect after checkin action

Closes #154

* Check for group access levels when getting the fields (#153)

* Consider field group state

Closes #155

* Added missing language strings

Closes #129

* JArrayHelper => ArrayHelper

* reduce model method (#160)

* Fixing permission inheritance to edit the value of a field in the form (#157)

* Correct path for parameters folder when not in Joomla libraries

* Type selection (#162)

Closes #134

* Display class attributes (#143)

* User home (#161)

* Add home parameter to media field

* Correct path for paramneters folder when not in Joomla libraries

* Fixing code style errors

* Add new context for com_contact for contact form fields (#115)

* Add new context for com_contact for contact form fields

* Prepare the value on com_contact mail

* Fixing travis code style errors

* Use the correct variable name

Closes #163

* Show the all label when no category is assigned

Closes #165

* Don't set 0 as default hint when not present

Closes #166

* Fix Sniffer Whitespace errors

* Fix Sniffer Whitespace errors (#167)

* Cleanup field model

* Support fields plugins

* Add gallery plugin

* Moved since tag to 3.7

* Some brushup

* Update controller.php

* Update field.php

* Update fields.php

* Update controller.php

* Update fields.php

* Update internal.php

* Update fields.php

* Update fields.xml

* Update field.php

* Update fields.php

* Update type.php

* Update section.php

* Update field.php

* Update field.xml

* Update filter_fields.xml

* Update field.php

* Update view.html.php

* Update edit.php

* Update modal_options.php

* Update view.html.php

* Update default.php

* Update default_batch_body.php

* Update modal.php

* Update field.xml

* Update en-GB.com_fields.sys.ini

* Update en-GB.com_fields.ini

* Update en-GB.plg_fields_gallery.ini

* Update en-GB.plg_fields_gallery.sys.ini

* Update en-GB.plg_system_fields.ini

* Update en-GB.plg_system_fields.sys.ini

* Update modal_article.php

* Update controller.php

* Update field.php

* Update default_custom.php

* Update captcha.php

* Update editor.php

* Update media.php

* Update user.php

* Update editor.xml

* Update media.xml

* Update user.xml

* Update usergrouplist.xml

* Update abstractlist.php

* Update domfieldinterface.php

* Update field.php

* Update email.php

* Update file.php

* Update imagelist.php

* Update sql.php

* Update tel.php

* Update textarea.php

* Update url.php

* Update calendar.xml

* Update checkboxes.xml

* Update imagelist.xml

* Update integer.xml

* Update list.xml

* Update radio.xml

* Update sql.xml

* Update textarea.xml

* Update url.xml

* Update gallery.xml

* plugin

* plugin2

* typo

* Change to uft8mb4

* Add new extensions to the script file

* Revert merge conflict composer changes

* No needed changes

* Library form fields should be unaware of com_fields

* Default to https

* Order the tables correct

* Cleanup language strings

* Code errors from cleanup

* Remove tags support

* Add category filter

* Display fielg group label when editing a group

* Display all fields which are assigned to all languages correct

* Don't show fields on categories as it is not ready yet

* Update fields.php

* Don't transofrm the category name

* Correct quote name function used

* Revert "Display fielg group label when editing a group"

This reverts commit 133133267e88a341cb5634d814a87e06f3e8daf3.

* Disable tags on custom field groups

* Don't include category state in filter on back end

* Remove debug code

* CS fix

* Shortening the line length

* Install SQL files for postgres

* Use JDatabaseQuery

* Option to show the user custom fields on the front on the contact view

* Comment for item_id table column

* Remove class for textareas

* Fix no menu item pages giving wrong links (#12020)

* [plg_system_logout] Load language files only when needed (#11736)

* Update logout.php

* cs and stuff

* Update logout.php

* Adding Options Button (#12033)

* fix paths (#12032)

* Fix routing for non-sef menu items with Modern Routing (#12021)

* [com_banners] - publishing time does not honor timezone (#11978)

* [com_banners ] -  publishing time does not honor timezone

* impress only when needed

impress only when needed

* CS fix

cs fix

* Order stylesheet attributes the same as link attributes (#12052)

* use joomla-projects docker

* Phase out JString calls in libraries (#12058)

* Admin app - JString -> StringHelper (#12056)

* Site app & Plugins - JString -> StringHelper (#12057)

* New Feature, Multilanguage: Add the possibility of displaying associations in article info and in articles list. Replaces #11935 (#12042)

* Replaces

* Modifying to fit Thomas proposal

* modifying fetching flags param

* adding showon + some

* [JAccess] Improve ACL asset preloading performance/memory consumption (#12028)

* improve component asset preload speed

* Update user.php

* Update access.php

* Update access.php

* Update access.php

* Update access.php

* not convinced yet, but revert changes in juser

* make sure that components are always preloaded

* only a logged users (with user id) can be root.

* [com_contact] - moved the captchaEnabled var from default to view (#11964)

* [com_contact] - moved the captchaEnabled var out of the default.php file to the view.html.php file

moved the captchaEnabled var out of the default.php file to the view.html.php file.

* moved captchaEnabled to the view

moved captchaEnabled to the view

* missed var  declarationi/initialization

missed var   $captchaSet  declarationi/initialization

* Session interaction refactoring (#10905)

* Allow lazy starting a session

* Defer starting sessions in the installer

* Defer starting session in JApplicationCms

* Deprecate creating sessions in JFactory

* Start restructuring session load sequence in JApplicationCms

* Add JSession object to onAfterSessionStart event trigger

* Override afterSessionStart for install app to not store the user

* Expand the database storage object for better API

* Fix session listener declaration

* Restructure session bootup and when the session metadata operations run

* 3.5 -> 3.6

* Remove option, out of scope

* Tweak things so the behavior is closer to current API

* Undefined variable

* Get store name from session object

* Remove this, out of scope

* PHPCS fix

* Deep rooted application dependencies are my absolute favorite

* Move comment, doc blocks, consistent method signature

* Again on the consistency, we don't like E_STRICT errors

* Nevermind, confused myself with checkSession, carry on...

* Until the day the unit tests no longer follow the doc block code styles...

* Lost line in merge

* One more merge conflict fix

* This internal coupling is addressed

* Fix query from merge

* [com_content] - add articles ordering by votes, ratings (#11225)

* Add setDocumentTitle() function in JViewLegacy (#11399)

* setDocumentTitle() function in JViewLegacy

* PHPCS

* JED turned to https (#12076)

* JED truned to https

JED truned to https
Minor fixes on proper use of Joomla!

* Update README.txt

Same as #12076

* JED turned https

* JED turned https

* JED turned https

* JED turned https

* JED turned https

* JED turned https

* JED turned https

* JED turned https

* JED turned https

* JED turned https

* JED turned https

* JED turned https

* JED turned https

* JED and community turned https

* JED, community and org turned https

* JED and community turned https

* JED and community turned https

* correct/add quotename (#12070)

* Label: Users Options: Login name > Username (#12073)

The language label "Change Login Name" is not consistent with the front-end label "Username".

This PR changes the language label to "Change Username" and the decription to "Allow users to change their Username when editing their profile."

The front-end login screens use the label "Username"
Login Component:
![login-component](https://cloud.githubusercontent.com/assets/1217850/18627996/579510ca-7e5e-11e6-9161-6268e9ccb213.png)

Login Module:
![login-module](https://cloud.githubusercontent.com/assets/1217850/18627997/579595ae-7e5e-11e6-8920-edf34631a3c7.png)

### Testing Instructions
#### Before the PR
Users > Manage > [Options] > [User Options]
The last entry is "Change Login Name" 
+ hover description "Allow users to change their Login name when editing their profile."

![login-back-end-options](https://cloud.githubusercontent.com/assets/1217850/18627995/5791ac32-7e5e-11e6-8b18-94a34372c422.png)


#### After the PR
Users > Manage > [Options] > [User Options]
The last entry is "Change Username" 
+ hover description "Allow users to change their Username when editing their profile."

![login-back-end-options-after](https://cloud.githubusercontent.com/assets/1217850/18627998/5799ceb2-7e5e-11e6-9a5e-234dcf2db14e.png)

* Update .travis.yml (#12079)

* Add vote order check

* [a11y] Protostar back to top (#12446)

* [a11y] Protostar - back to top link

* Oops Andre was right

* add anchor for non-js enabled browsers

* Added missing com_fields component from last merge

Commit:
Replace Platform version numbers in libraries/legacy (#12242)

Commit:
Remove variable declaration (Fix #12619)

brianteeman committed:
Lang corrections post custom fields merge (#12618)

* alpha sort new string

* alpha sort new strings

* alpha sort strings

* alpha sort again

* typos and code style

* Backend and frontend en-GB.lib_joomla.ini should be exactly the same.

* alpha sort

Commit:
Change typehint for JControllerBase to be more permissive (#12650)

Fedik committed:
Fix for Module Positions inaccessible wit Hathor admin Template (#12621)

* Fix for Module Positions inaccessable wit Hathor admin Template

* Remove chosen.js from Module Positions field

infograf768 committed:
com_fields : checking for availability of component, correcting 2 fatal errors (#12646)

* com_fields : checking for availability of component, correcting 2 fatal
errors

* Simplifying checks + others

zero-24 committed:
[fix] When installing Joomla! and mail sending fails you are not getting to the next screen. (#11895)

* fix error if mail send fails and set it as error

* send message if send fails

wilsonge committed:
Remove deprecated function use from com_languages (#12188)

photodude committed:
Fix DSN seperator (#12667)

We use semicolons `;` everywhere else we should here as well.

bertmert committed:
com_fields code style (missing line break) (#12669)

* CS (line break)

* CS (line break)

okonomiyaki3000 committed:
CodeMirror version 5.20.2 (#12541)

brianteeman committed:
recompile less using our generatecss.php script (#12548)

bpeterson69 committed:
Update uri.php (#8690)

* Update uri.php

Added additional validation code to support reverse proxies.  Without this additional validation, a Joomla site set to force SSL on the entire site will go into an infinite redirect loop because the reverse proxy is sending the information to confirm SSL in the $_SERVER variable key 'HTTP_X_FORWARDED_PROTO', not the key 'HTTPS'.

* Update uri.php

* Update uri.php

* Update uri.php

* Update uri.php

Commit:
Remove useless empty rule (#12678)

laoneo committed:
Remove not needed version and hits field from db schema (#12674)

alex7r committed:
Fix PHPDoc (#12686)

Fixing wrong arg type in PHPDoc

coolcat-creations committed:
Add more semantic html5 elements to module tag options (#12455)

* Add more semantic html5 elements to module tag options

* Add back address tag

* Add blank line before comment

zero-24 committed:
Set correct default values for user creation in the backend (#12589)

* Update joomla.sql

* Update joomla.sql

* Update joomla.sql

brianteeman committed:
enable custom fields option (#12663)

laoneo committed:
Check if multilanguage is enabled (#12675)

laoneo committed:
Remove obsolete modals (#12672)

brianteeman committed:
readme download links (#12702)

* readme download links

* readme download links (reverted from commit 382bae3d89375beebc602bc273aee8f6382359f7)

* correct

andrepereiradasilva committed:
Add script version to all js files and replace deprecated script signatures usage (#12373)

* backend templates

* frontend templates

* plugins

* frontend modules

* layouts

* installation

* tests

* tests

* frontend components

* libraries

* administrator components

* ups remove duplicated script

* [a11y] Protostar back to top (#12446)

* [a11y] Protostar - back to top link

* Oops Andre was right

* add anchor for non-js enabled browsers

* some leftovers after recent merges

* yet some more

* plg_fields_gallery

* removed wrong file

* remove empty line

* add version to installation

* fix for beez3

brianteeman committed:
green for recommended setting (#12709)

alikon committed:
sql - support for scss,sass source formats (#12712)

* support for scss,sass source formats

https://github.com/joomla/joomla-cms/pull/10186#issuecomment-257952879

* support for scss,sass source formats

* support for scss,sass source formats 

support for scss,sass source formats

* support for scss,sass source formats 

support for scss,sass source formats

* support for scss,sass source formats 

support for scss,sass source formats

* support for scss,sass source formats 

support for scss,sass source formats

* support for scss,sass source formats 

support for scss,sass source formats

* support for scss,sass source formats 

support for scss,sass source formats

* remove the full param update

* remove the full param update

* remove the full param update

zero-24 committed:
Updates for the new downloads Portal (#12707)

* move to downloads.joomla.org

* one more

brianteeman committed:
shorten label (#12708)

andrepereiradasilva committed:
Don't add media version if media version is empty (#12716)

* don't add media version if it's empty

* Update head.php

andrepereiradasilva committed:
[com_fields fields view] SQL error on ordering  (#12718)

* Update default.php

* state!

* still missing here

* we still need it here or no ordering icon

andrepereiradasilva committed:
Update default.php (#12717)

Commit:
Use JComponentHelper::getParams() (#12723)

[com_fields] Use JComponentHelper::getParams() instead of JComponentHelper::getComponent('com_foo')->params

ciar4n committed:
Global help refresh padding fix (#12722)

ggppdk committed:
Fix double parameters creation for parameters of home template and of template override (#12688)

* Fix double parameters creation for parameters of home style

* Fixed double creation of parameters for template override

andrepereiradasilva committed:
add mediaversion to polyfill (#12714)

brianteeman committed:
add support for user.js to beez and protostar (#12684)

* add support for user.js to beez and protostar

* fix missing bracket

* Update index.php

* Update index.php

* Update offline.php

* fix for beez3

andrepereiradasilva committed:
Add version and IE conditional statements to JHtml::stylesheet/JDocument::addStyleSheet methods and deprecate JDocument::addSyleSheetVersion() (#12375)

* let's do the same for stylesheets

* cs

* ups

* in the previou ssignature the second parameter was an array!

* unit tests

* don't force array

* more unit tests corrections

* more unit tests adjustments

* the array must be equal ...

andrepereiradasilva committed:
Make js ajax error messages more useful when using Joomla.request (#12437)

* improve ajaxErrorsMessages treatment

* improvements

Commit:
Tab state logic fix (#12503)

* fix

* regex improve

* regex impr

* regex again

* remove limitation for id being last param

andrepereiradasilva committed:
Use the new native title metadata in installed languages view (#12641)

* use the new native title

* this is not needed now

* further code improvements and bug corrections

* we don't need this also now

* Add -

infograf768 committed:
com_feeds: normalising, adding strings (#12653)

andrepereiradasilva committed:
Media Version on installation (solves #12711) (#12713)

* mediaversion on installation

* Update index.php

brianteeman committed:
[com_fields]remove captcha from field types (#12719)

* remove captcha from field types

* remove captcha from field types (reverted from commit 70c43c55e48b99b80b7946d280c16cd07b037096)

* different approach

Commit:
Use proper field name to show modified user (#12736)

brianteeman committed:
Add profile link to login module (#12539)

* Add link to module

* add language strings

* add param to the admin

* enabled param check in module

* update beez template override

infograf768 committed:
Undesired Notices when publishing child category (#12737)

andrepereiradasilva committed:
Remove showon from global config cache fields (#12742)

* Update application.xml

* Update application.xml

* Update application.xml

* system cache

andrepereiradasilva committed:
Add stylesheet version to all css files and replace deprecated stylesheet signatures usage (#12720)

* system site template

* protostar site template

* beez3 site template

* beez3 site template mod_languages

* installation template

* layouts

* libraries

* site components

* isis admin template

* hathor admin template

* hathor override

* administrator components

* beez3 are not relative

* missed home

* some more missing

* correct paths

brianteeman committed:
Alternative description (#12677)

* Alternative description

* correct copy paste error

* sync

Devportobello committed:
[NF] Menu item visibility indicator (#11998)

* Menuitem visibility indicator

* cs test1

* cs test2

* ux

* moveLabelAfterAlias

* handleInvalidJson

* cs

Commit:
JFormFieldList shows the global values if present. (#11911)

* JFormFieldList shows the global values if present.

* Moving string to en-GB.ini (where JGLOBAL_USE_GLOBAL lives)

* Rewrite to include the global value in the option itself.
Introduce "useglobal" attribute for listfield.

* Show a notice when global values aren't found

* Removed robots from being looked up.
Notice is now generic and will only show once.

* Adding global values to menu items

* Changed menuitems for all com_content views

* Single Article Menu Item done.

* Adding default value to some text fields

* global values for featured view

* Rebased and moved code to JFormAbstractlist to solve conflicts with com_fields PR

* Added useglobal to some newly introduced parameters.

* Codestyle

wilsonge committed:
Make category filters form wcag 2.0 Accessibility compliant. Fixes #8599

zero-24 committed:
[3.7] Add a Postinstall Message to warn users on nearly unsupported envoirments (#12237)

* postinstall_joomla4

* postinstall_joomla4

* remove temp file

* language strings

* install sql

Commit:
Fix front end XTD menus and contacts buttons (#12321)

* menus

* cs

* xtd-contacts

* session check, correct version

* articles

* cs identation fix

* Update modules.php

solving issue for Manager enable to use xtd module in back-end

* Update router.php

Solving router issue and let xtd article work fine

* Update router.php

Solving contact router issue

* Filters and menu sql fixes

* unification

* implode empty array

* fixes

* revert router, add lang, add/change ACL

* Update router.php

* Update router.php

brianteeman committed:
Improve sef plugin tooltip (#12472)

roland-d committed:
Fix the broken search (#12750)

brianteeman committed:
copy paste error in docblock (#12757)

andrepereiradasilva committed:
[installation] Clean assets rules in sql files (#12069)

* clean assest rules in sql files

* Update sample_blog.sql

* Update sample_learn.sql

* Update sample_testing.sql

* Update sample_blog.sql

* Update sample_learn.sql

* Update sample_testing.sql

* Update sample_learn.sql

* Update sample_testing.sql

Commit:
Remove unneeded metadata.xml files (#11990)

* Remove unneeded metadata.xml files

* Remove files during update

brianteeman committed:
com_search on back-end: Results links opens in backend window instead of new Tab (#12626)

* Updated installation language files

* 2fa handeling for mcrypt and openssl (#12497)

* handeling for mcrypt and openssl

* Language changes suggested bei Brian

* search results in new window

* 2fa handeling for mcrypt and openssl (#12497)

* handeling for mcrypt and openssl

* Language changes suggested bei Brian
 (reverted from commit 08d6728b6e75ab763e920edffeedc8b209228ec1)

* Updated installation language files (reverted from commit e504f715b2093560a16c1488000a036c80c51ae8)

* language string

Hackwar committed:
Fixing Contact urls for linked authors (New) (#12629)

* Fixing URLs in plg_content_contact

* Replacing require_once with class loader

Commit:
Fix notices in contact view (#12729)

Commit:
Move fields label to first page (#12740)

brianteeman committed:
redo popover fix (#12749)

izharaazmi committed:
Make Invalid Token warnings more user friendly (#8445)

* The session token validation implementation currently gives an empty white screen with "Invalid token" text. This is total uninformative to a normal user.
This update implicitly redirects to the previous page if the token is not valid. To prevent implicit redirect pass `false` as the second argument to `JSession::validateToken()`

To make use of this method we need to replace any occurrence of `JSession::checkToken('post') or jexit(JText::_('JINVALID_TOKEN'));` with `JSession::validateToken('post');`
Example is set up in com_users > login method.

* Moved the checkToken method to JControllerLegacy as it is the more closer context to the usage.
This allows implicit redirect to the REFERER page in case of token mismatch. In all other case it is same as calling JSession::checkToken() directly.

Changes made in the frontend components to make this change test-able.

Note: We intend to discourage the `die` usage everywhere which leads the the said uninformative white page.

* Fix wrong function call

* use automatic since tag macro

Thanks @andrepereiradasilva

zero-24 committed:
one clean line to much (#12764)

brianteeman committed:
Abbreviate Miscellaneous

PhilETaylor committed:
Add 4.0-dev branch to CONTRIBUTING.md (#12774)

* Add 4.0-dev branch to CONTRIBUTING.md


Add 4.0-dev branch to CONTRIBUTING.md

* -in-compatible  +incompatible

jeckodevelopment committed:
Update CONTRIBUTING.md (#12776)

Removed 2.5 branch

PhilETaylor committed:
Update sample_brochure.sql (#12771)

PhilETaylor committed:
Fix typo in sample_brochure.sql (#12770)

- This is the latest new from us
+ This is the latest news from us

brianteeman committed:
logical message (#12767)

jeckodevelopment committed:
Merge pull request #12769 from brianteeman/miscellaneous

Abbreviate Miscellaneous in Smart Search options

andrepereiradasilva committed:
Update head.php (#12781)

andrepereiradasilva committed:
Update default.php (#12766)

brianteeman committed:
correct data in installation sql (#12760)

* correct data in installation sql

* oops

* last one

* corrections

laoneo committed:
Change key to name as for lists (#12698)

ggppdk committed:
[com_fields] Fix text-value being reverse for fields: list, radio, checkbox. Fix fields: radio, checkboxes not showing (#12739)

* Fix value-text being reverse for fields list, radio, checkbox, change 'name','key' label to be 'text' for these fields

* Fixed calling method from parent class

* Added missing layout file for radio field

* Fixed character '0' unusuable as field value

* Revert fix for zero character as value

* Proper variable names for view layouts of options-using fields

* They are options not just values

* JText the Text of options in the view layouts, since respective form fields do it already

Commit:
Remove an unused line. (#12808)

andrepereiradasilva committed:
Remove error message from keepalive behaviour (#12512)

* remove error message from keepalive

* minified version

brianteeman committed:
[com_fields] add aliastag to the blacklist (#12779)

* add aliastag to the blacklist to prevent being loaded in com_fields as an available fieldtype

* Update type.php

rdeutz committed:
fix merge conficts for #8941

jeckodevelopment committed:
Removing 3.6 from Contributing (#12823)

Removed:
| 3.6.x | Branch for the next minor Joomla version. New backward compatible features go into this branch. Commits to staging will be applied to this branch as well. |

Removed the above note from the branches table since as specified in #12775 the 3.7 branch will be removed asap and the staging branch now is used for the 3.7 minor.

brianteeman committed:
its a super user not an administrator (#12756)

* its a super user not an administrator

* add title

andrepereiradasilva committed:
Solve warning in with JHtml::includeRelativeFiles (#12777) (#12780)

* Update html.php

* missed the dot

* add strlen more than 4 check

Commit:
JFormAbstractlist ignores 0 as option text (#12814)

Commit:
Updater follows Location headers for valid archive filename (#12817)

* made download() follow Location headers to resolve the valid filename of the update archive

* integrated suggestion of ggppdk, valid behavior if no query string

* strict type checking

* make Travis happy again... code style error

andrepereiradasilva committed:
[installation] Created in 2011-01-01? Update created date plus minor corrections (#11577)

* updated created date plus minor corrections

* remove empty line

* exclude sample dat testing

* fix conflicts 1

* fix conflicts and update all other date tables

* cs

* remove checked_out dates

* fix conflicts

* Update database.php

* fix conflicts 2

* fix conflicts final

Commit:
Updated installation language files (ga-IE)

infograf768 committed:
Email address: authorise single quote in local-part (#12835)

andrepereiradasilva committed:
Add consistency in Permissions tabs (#12851)

* Permissions tab in global config

* com_modules Permissions tab translation

* new/edit module permissions tab

* change menus too

andrepereiradasilva committed:
More useful message when "Force HTTPS" config save fails (because of HTTPS config issues) (#12465)

* show proper error

* add/change messages

smehrbrodt committed:
[com_fields] Improve wording of some messages (#12856)

* Update en-GB.plg_system_fields.sys.ini

* Update en-GB.plg_system_fields.ini

hordijk committed:
Update sef.php (#5741)

Changed the reqex for all unknown protocols.

Removed the + character for the whitespace to solve a max_execution_timeout exception.
If you have a large output document the regex '#\s+(src|href|poster)="(?!/|' . $protocols . '|\#|\')([^"]*)"#m' takes way to long. Supporting only one \s will do the trick as well. The result will be additional spaces between attributes, but this shouldn't harm.

bembelimen committed:
Don't skip JHtml's bootstrap call for override check (#12854)

shur committed:
Delete unnecesary catslug query from mod_related_items (#8946)

* Delete unnecesary catslug from mod_related_items

* remove whitespace

* catslug depricated note

* start comment with a capital letter

* add blank line

* Update helper.php

* Update helper.php

shur committed:
Delete unnecesary catslug query from mod_related_items (#8946)

* Delete unnecesary catslug from mod_related_items

* remove whitespace

* catslug depricated note

* start comment with a capital letter

* add blank line

* Update helper.php

* Update helper.php

roland-d committed:
Merge branch 'nested-form' into staging

* nested-form:
  Delete unnecesary catslug query from mod_related_items (#8946)

infograf768 committed:
3.7.0 : Correcting login/logout lang strings (#12337)

* Correcting login/logout lang strings

* Updating strings

andrepereiradasilva committed:
Update item.php (#12825)

Commit:
Removing the notice when a global value is not found. (#12858)

infograf768 committed:
com_fields: allowing custom field Label to be translated by a string (#12656)

* com_fields: Allow translation of field label via language string

* better this way

zero-24 commented:
Any news or should this be closed?

brianteeman commented:
I suspect that others like me and we don't understand what it is you are doing here

mbabker commented:
To use the reinstall feature in the update component, right now it requires that the update server serve the full upgrade package to all sites (the smaller patch packages therefore can't be used).  This would apparently allow for the patch packages to be served while using the upgrade packages for reinstall by using a new XML element specific to the update component (so specific it's not even introduced into the core update processor's library).

Honestly, I don't see this being useful before 4.0.  It's only going to increase maintenance of the XML manifests (for example 3.6 will still have to have the upgrade packages but 3.7, if merged, would allow the reinstall logic).

zero-24 commented:
Currently we have two key features of the updater. Update joomla and reinstall the core files.

For the update we can use a small package e.g. 3.6.x to 3.6.5 for the reinstall we need the full package With all files. Currently we can only add one download file that is used for both features so any update runs via the full package.

With this patch (or similiar) we can use a different URL for reinstall (the full package) and for update (the smaler package)

Is this clearer now? If not please let me know ;)

zero-24 commented:
> Honestly, I don't see this being useful before 4.0

So this can /should be closed?

mbabker commented:
I'm just one person with an opinion 😉

wojsmol commented:
@zero-24 Please rebase this PR - then I will test this.

infograf768 committed:
Updating Hathor to use new Live and Upload for joomlaupdate (#12454)

* Uodating Hathor to use new Live and Upload for joomlaupdate

* Changing float of #form-login

* Adding min-height

infograf768 committed:
Using a neutral colour for multilingual buttons at installation time (#12879)

* Using a neutral colour for multilingual buttons at installation time

* Using a darker shade of blue

* Reduce CSS, using wildcard

brianteeman committed:
com_newsfeed update tooltip (#12892)

* update tooltip

* update in the item view options

laoneo committed:
If the label is empty set it from the title automatically (#12885)

stutteringp0et committed:
JApplicationWeb::setHeader/sendHeaders behavior (#9836)

* Update web.php

setHeader $replace previously did nothing other than delete a value - even when false it allowed a duplicate to be created which effectively replaces the original.

This modification searches for an existing header with the same name and only inserts it if it doesn't already exist, or if $replace is true.

* JApplicationWeb::setHeader behavior

Fix Travis CI error regarding invalid foreach

* JApplicationWeb::setHeader behavior

Formatting changes

* JApplicationWeb::setHeader behavior

Formatting issues

* Fix testSetHeader



in line 1314 "$this->class->setHeader('foo', 'car');", the method is called without the $replace argument, meaning it uses the default of "false".

The way headers are set, allowing a duplicate here means the header is replaced in JApplicationWeb::header.  If $replace isn't enforced in setHeader then duplicates mean the header gets replaced anyway.

The attempted entry is removed from the assertEquals test.

* JApplicationWeb -setHeader and sendHeader upgrade

Current Joomla header operation causes a duplicate header to replace an original header even when $replace = false.  The replacement is a result of interaction between setHeader and sendHeader.

setHeader happily adds duplicate entries when $replace = false
sendHeader happily replaces headers because its default is to replace

This commit alters both:

setHeader now checks an internal list of headers which may only have one value.
If the header is a single value only, the header array is updated only if $replace = true and the update is ignored otherwise
If the header is not a single value, the header is replaced if $replace = true, otherwise it is added as a multiple value 

sendHeader now creates a new array of values, aggregating values under the header name.  When the header is sent to the header() method, its value is imploded on ', '

* Returning test to original state

this should be removed from the branch but I'm not sure how to do it.

* JApplicationWeb::setHeader behavior

Update to fix status header

* JApplicationWeb -setHeader and sendHeader upgrade

Fixing for test output

* JApplicationWeb -setHeader and sendHeader upgrade

formatting

* JApplicationWeb -setHeader and sendHeader upgrade

formatting

* JApplicationWeb -setHeader and sendHeader upgrade

formatting

* JApplicationWeb -setHeader and sendHeader upgrade

refinement - removed unnecessary parens

* JApplicationWeb -setHeader and sendHeader upgrade

performance enhancement

* JApplicationWeb -setHeader and sendHeader upgrade

performance enhancements, 2 fewer foreach iterations for the headers

* JApplicationWeb -setHeader and sendHeader upgrade

formatting fix

* JApplicationWeb -setHeader and sendHeader upgrade

Missed a variable change i sendHeader

* JApplicationWeb -setHeader and sendHeader upgrade

formatting fix

* Update web.php

* JApplicationWeb -setHeader and sendHeader upgrade

fix formatting

* JApplicationWeb -setHeader and sendHeader upgrade

Adding ability to unset (remove) an existing header using $replace=true and $value=null

setHeader('name', null, true) now removes the header entirely, where previous behavior would have sent the header with an empty value.

* JApplicationWeb -setHeader and sendHeader upgrade

formatting fix

laoneo committed:
Resize to the correct height of the iframe (#11554)

* Get the correct height of the iframe

* Making it more robust for all browsers

ciar4n committed:
Subform.Repeatable styling (#12634)

Commit:
Add Tag filter (#12640)

* Add Tag filter

* Codestyle

* add string

* Updating XML file to use new string

andrepereiradasilva committed:
Use new JLanguageHelper::getInstalledLanguages across core (#12671)

* new JLanguageHelper::getInstalledLanguages

* some mroe improvs

* correct deprecated comment

* fix bug

* cs

* positive

* strict

* be sure there in an index for each client

* check if file exists

* remove extra )

* remove check

chrisdavenport committed:
[Smart Search] Log peak memory usage in web indexer (#12680)

laoneo committed:
Remove not used option (#12699)

chrisdavenport committed:
[Smart Search] Fix for tags issue #10122 (#12745)

pe7er committed:
database table #__extensions - Set enabled to 0 by default (#12753)

* extension databasetable - set enabled to 0 by default

* database update for existing sites

* database update for mysql for existing sites

andrepereiradasilva committed:
the component name is com_contact not com_contacts (#12797)

andrepereiradasilva committed:
Update view.html.php (#12800)

andrepereiradasilva committed:
[ACL] [com_menus items view] Correct check when "All Menus" is selected (#12815)

* add menu type id to the model so we can use it on acl checks

* add the correct acl check

andrepereiradasilva committed:
[ACL] [new category] Correct invalid core.edit.state check when creating a new category (#12820)

* Update category.php

* Update category.php

andrepereiradasilva committed:
Update groupparent.php (#12824)

andrepereiradasilva committed:
[ACL] [com_fields fields view] Correct when display batch button (#12826)

* Update view.html.php

* Update default.php

andrepereiradasilva committed:
Update field.php (#12827)

brianteeman committed:
google authenticator plugin description (#12833)

* make google authenticator a link and add freeotp as an example

* add intro

* remove google

* another freeotp link

andrepereiradasilva committed:
Update document.php (#12836)

andrepereiradasilva committed:
com_users notes category incorrect ACL level in several installation sql (#12852)

* Update joomla.sql

* Update sample_blog.sql

* Update sample_brochure.sql

* Update sample_data.sql

* Update sample_data.sql

* Update sample_blog.sql

* Update joomla.sql

* Update sample_data.sql

* Update sample_brochure.sql

* Update sample_blog.sql

* Update joomla.sql

Commit:
Remove Searchbox from all fields except Category (#12855)

* Remove Searchbox from all fields except Category

* Missed module edit form

Commit:
Add "show global value" feature to com_contact (#12861)

* Add "show global value" feature to com_contact

* Missed text fields

* Reverting order of show_info options in com_contact component config.

Commit:
Add "show global value" feature to com_content frontend edit (#12863)

* Add "show global value" feature to com_content frontend edit

* Enable show global value for "Show Associations"

Commit:
Enable show global value for "Show Associations" (#12864)

Commit:
Add "show global value" feature to com_tags (#12865)

* Add "show global value" feature to com_tags

* Show global value for "Maximum Characters"

Commit:
Adding a "Use Global" option to tag list menu item (#12868)

infograf768 committed:
Menu item alias: preventing saving forbidden aliases (#12871)

andrepereiradasilva committed:
[unit tests] Remove empty acl rules (#12915)

* clean unit tests assets table

* Update jos_assets_composite.csv

* jaccesstest

brianteeman committed:
typo (#12913)

Commit:
Removing deprecated methods and add missing language key (#12899)

Also minor codestyle

Commit:
Adding the ability to use the global value for character count in newsfeeds (#12869)

* Adding the ability to use the global value for character count in newsfeeds

* Missing views

andrepereiradasilva committed:
Update default.xml (#12917)

andrepereiradasilva committed:
Update edit.php (#12818)

andrepereiradasilva committed:
[JHtmlNumber::bytes] Format number according to language (#12929)

* return translated number

* ups

* cs

* fix unit tests

* fix last unit tests

* minor cs

chrBrd committed:
JSession patched to set session _state to 'inactive' when session is closed. (#12928)

810 committed:
remove tab on meta charset (#12895)

* remove tab on meta charset

* Update code

PhilETaylor committed:
Fix for Undefined offset in Content History preview popup (#12791)

* Fix Undefined offset

* tabs/spacing

* If someone has faked an article id - dont show trace, show an error

* make travis happy

* make travis happy

* travis I hate you

* tab/space hell

* tab/spaces hell

* changes as requested

* remove the $language var as per comment

tecpromotion committed:
Set correct component id for system links (#12938)

* Update menu.php

* Update item.php

ciar4n committed:
a11y regression fix (#12935)

zero-24 committed:
merge conflicts
